static/panes/gccdump-view.ts

```diff
@@ -0,0 +1,423 @@
+        this.optionBlocksTitle = this.optionBlocksButton.prop('title');
+
+        this.optionVopsButton = this.domRoot.find("[data-bind='vopsOption']");
+        this.optionVopsTitle = this.optionVopsButton.prop('title');
+
+        this.optionLinenoButton = this.domRoot.find("[data-bind='linenoOption']");
+        this.optionLinenoTitle = this.optionLinenoButton.prop('title');
+
+        this.optionUidButton = this.domRoot.find("[data-bind='uidOption']");
+        this.optionUidTitle = this.optionUidButton.prop('title');
+
+        this.optionAllButton = this.domRoot.find("[data-bind='allOption']");
+        this.optionAllTitle = this.optionAllButton.prop('title');
+    }
+
+    registerCallbacks() {
+        this.filters.on('change', this.onFilterChange.bind(this));
+        this.selectize.on('change', this.onPassSelect.bind(this));
+
+        this.eventHub.emit('gccDumpViewOpened', this.compilerInfo.compilerId);
+        this.eventHub.emit('requestSettings');
+
+        this.container.on('resize', this.resize, this);
+        this.container.on('shown', this.resize, this);
+
+        this.cursorSelectionThrottledFunction = _.throttle(this.onDidChangeCursorSelection.bind(this), 500);
+        this.editor.onDidChangeCursorSelection(e => {
+            this.cursorSelectionThrottledFunction(e);
+        });
+    }
+
+    updateButtons() {
+        const formatButtonTitle = (button, title) =>
+            button.prop('title', '[' + (button.hasClass('active') ? 'ON' : 'OFF') + '] ' + title);
+        formatButtonTitle(this.dumpTreesButton, this.dumpTreesTitle);
+        formatButtonTitle(this.dumpRtlButton, this.dumpRtlTitle);
+        formatButtonTitle(this.dumpIpaButton, this.dumpIpaTitle);
+        formatButtonTitle(this.optionAddressButton, this.optionAddressTitle);
+        formatButtonTitle(this.optionSlimButton, this.optionSlimTitle);
+        formatButtonTitle(this.optionRawButton, this.optionRawTitle);
+        formatButtonTitle(this.optionDetailsButton, this.optionDetailsTitle);
+        formatButtonTitle(this.optionStatsButton, this.optionStatsTitle);
+        formatButtonTitle(this.optionBlocksButton, this.optionBlocksTitle);
+        formatButtonTitle(this.optionVopsButton, this.optionVopsTitle);
+        formatButtonTitle(this.optionLinenoButton, this.optionLinenoTitle);
+        formatButtonTitle(this.optionUidButton, this.optionUidTitle);
+        formatButtonTitle(this.optionAllButton, this.optionAllTitle);
+    }
+
+    // Disable view's menu when invalid compiler has been
+    // selected after view is opened.
+    onUiNotReady() {
+        // disable drop down menu and buttons
+        this.selectize.disable();
+        this.dumpFiltersButtons.prop('disabled', true);
+    }
+
+    onUiReady() {
+        // enable drop down menu and buttons
+        this.selectize.enable();
+
+        this.dumpFiltersButtons.prop('disabled', false);
+    }
+
+    onPassSelect(passId: never) {
```

Comment:
What's the reason for `never` here?

Reply:
Ooops I'd meant this as a note to myself to come back and enter the correct type, I'm actually surprised ts let it compile 🤔